### PR TITLE
net-misc/chrony: Privilege separation support, enable seccomp

### PR DESCRIPTION
**Pull request by thesamesam**
- When caps is enabled, drop to the user ntp (acct-user/ntp), as opposed to remaining root.
   (Adds a tmpfile.d entry for /run/chrony to ensure correct permissions.)

- We already have USE=seccomp but chronyd won't do anything unless -F is set to 1. We could also set -F -1 which will log any syscalls which would've been blocked but won't deny them.

Closes: https://bugs.gentoo.org/711058
Signed-off-by: Sam James (sam_c) <sam@cmpct.info>

### CODE REVIEW

File: net-misc/chrony/chrony-3.5-r3.ebuild [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+DESCRIPTION="NTP client and server programs"
+HOMEPAGE="https://chrony.tuxfamily.org/"
+SRC_URI="https://download.tuxfamily.org/${PN}/${P/_/-}.tar.gz"
+LICENSE="GPL-2"
+SLOT="0"
+
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ppc ~ppc64 ~sparc ~x86"
+IUSE="
+	+adns caps +cmdmon html ipv6 libedit +ntp +phc pps readline +refclock +rtc
```
Comment (Whissi) [inline net-misc/chrony/chrony-3.5-r3.ebuild]: Don't you want to enable either USE=caps or USE=seccomp by default?

File: net-misc/chrony/chrony-3.5-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	newinitd "${FILESDIR}"/chronyd.init-r2 chronyd
+	newconfd "${T}"/chronyd.conf chronyd
+
+	insinto /etc/${PN}
+	newins examples/chrony.conf.example1 chrony.conf
+
+	docinto examples
+	dodoc examples/*.example*
+
+	newtmpfiles - chronyd.conf <<<"d /run/chrony 0750 $(usex caps 'ntp ntp' 'root root')"
```
Comment (Whissi) [inline net-misc/chrony/chrony-3.5-r3.ebuild]: If /run/chrony must exists before chronyd is able to start, you should call tmpfiles_process in pkg_postinst.

File: net-misc/chrony/chrony-3.5-r3.ebuild [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}"/chrony-2.4-r1.logrotate chrony
+
+	systemd_dounit examples/{chronyd,chrony-wait}.service
+	systemd_enable_ntpunit 50-chrony chronyd.service
+}
+
+pkg_preinst() {
+	if use caps && has_version net-misc/chrony[-caps]; then
```
Comment (Whissi) [inline net-misc/chrony/chrony-3.5-r3.ebuild]: To be honest, I don't know what tmpfiles_process will do if directory already exists but with different permissions. I expect that we can get rid of the elog. But this must be verified.

File: net-misc/chrony/chrony-3.5-r3.ebuild (hunk 1 above)
Comment (thesamesam) [reply net-misc/chrony/chrony-3.5-r3.ebuild]: Yes, I was being a bit cautious but it should be fine (especially caps).

File: net-misc/chrony/chrony-3.5-r3.ebuild (hunk 2 above)
Comment (thesamesam) [reply net-misc/chrony/chrony-3.5-r3.ebuild]: You're right, I've done some research into the phases and this is obvious now!

File: net-misc/chrony/chrony-3.5-r3.ebuild (hunk 3 above)
Comment (thesamesam) [reply net-misc/chrony/chrony-3.5-r3.ebuild]: I've checked and it overrides the previous permissions, so let's kill it.

**Pull request closed without merging**